MFEM requires C++17 without CMake cxx_std_17

Hi y'all,

I recently upgraded to MFEM 4.9 and was hit with a build issue:

```
fem/dfem/util.hpp:553:12: error: no template named 'variant' in namespace 'std'
```

This is on macOS with AppleClang. I like that C++17 features are being used, but the requirement is not being propagated via CMake. To fix this, I added the following line to my CMakeLists.txt:

```
target_compile_features(mfem INTERFACE cxx_std_17)
```

Ideally, a similar line should be added to the MFEM CMakeLists.txt file so that the C++17 requirement is automatically communicated to any projects which use `find_package(MFEM)`. The `CMAKE_CXX_STANDARD` method only works for internal project code.

Thank you,
Aiden

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -723,6 +723,7 @@ set(MFEM_INSTALL_DIR ${CMAKE_INSTALL_PREFIX})
 
 # Declaring the library
 mfem_add_library(mfem ${SOURCES} ${HEADERS} ${MASTER_HEADERS})
+target_compile_features(mfem PUBLIC cxx_std_${CMAKE_CXX_STANDARD})
 # message(STATUS "TPL_LIBRARIES = ${TPL_LIBRARIES}")
 target_link_libraries(mfem PUBLIC ${TPL_LIBRARIES} ${TPL_TARGETS})
 if (TPL_TARGETS)

